src/createApp.ts

```diff
@@ -837,6 +852,48 @@ const createApp = compose({
 			return instanceRegistries.get(this).identifyStore(store);
 		},
 
+		createWidget<U extends Child, O>(this: App, factory: ComposeFactory<U, O>, options: any = {}): Promise<[ string, U ]> {
+			const app = this;
+			const { defaultWidgetStore, registryProvider } = this;
+			const { id = generateWidgetId() } = options;
+			// Like for custom elements, don't add the generated ID to the options.
+
+			function configureWidget(): Promise<any> | void {
+				// Ensure no other widget with this ID exists.
+				if (publicRegistries.get(app).hasWidget(id)) {
+					return Promise.reject(new Error(`A widget with ID '${id}' already exists`));
+				}
+
+				if (!options.registryProvider) {
+					options.registryProvider = registryProvider;
+				}
+
+				if (options.id && !options.stateFrom && defaultWidgetStore) {
```

Comment:
Yeah, I was thinking about this... there is no harm in doing it, but it will "always" do it and I don't know if that then becomes surprising an issue...  Because it isn't injecting the store into the mix, doing this has side-effects of which the initiator does not have control of, where if it was not present and having some way of making it optional just complicates things needlessly... so are we fine with it always happening?